utils/build-script-impl

```diff
@@ -2267,146 +2531,178 @@ for deployment_target in "${HOST_TARGET}" "${CROSS_COMPILE_TOOLS_DEPLOYMENT_TARG
         esac
 
         if [[ -z "${INSTALL_DESTDIR}" ]] ; then
-            echo "error: --install-destdir is required"
+            echo "--install-destdir is required to install products."
             exit 1
         fi
 
-        if [ "${CROSS_COMPILE_TOOLS_DEPLOYMENT_TARGETS}" ] ; then
-            # If cross compiling tools, install into a deployment target specific subdirectory.
-            if [[ ! "${SKIP_MERGE_LIPO_CROSS_COMPILE_TOOLS}" ]] ; then
-                target_install_destdir="${BUILD_DIR}"/intermediate-install/"${deployment_target}"
-            else
-                target_install_destdir="${INSTALL_DESTDIR}"/"${deployment_target}"
-            fi
-        else
-            target_install_destdir="${INSTALL_DESTDIR}"
-        fi
-
         echo "--- Installing ${product} ---"
-        build_dir=$(build_directory ${deployment_target} ${product})
+        build_dir=$(build_directory ${host} ${product})
 
         set -x
 
-        DESTDIR="${target_install_destdir}" "${CMAKE}" --build "${build_dir}" -- ${INSTALL_TARGETS}
+        DESTDIR="${host_install_destdir}" "${CMAKE}" --build "${build_dir}" -- ${INSTALL_TARGETS}
         { set +x; } 2>/dev/null
     done
 
+    if [[ "${DARWIN_INSTALL_EXTRACT_SYMBOLS}" ]] && [[ $(host_has_darwin_symbols ${host}) ]]; then
+        set -x
+        # Copy executables and shared libraries from the `host_install_destdir` to
+        # INSTALL_SYMROOT and run dsymutil on them.
+        (cd "${host_install_destdir}" &&
+         find ./ -perm -0111 -type f -print | cpio -pdm "${INSTALL_SYMROOT}")
+
+        # Run dsymutil on executables and shared libraries.
+        #
+        # Exclude shell scripts.
+        (cd "${INSTALL_SYMROOT}" &&
+         find ./ -perm -0111 -type f -print | \
+           grep -v swift-stdlib-tool | \
+           grep -v crashlog.py | \
+           grep -v symbolication.py | \
+           xargs -n 1 -P ${BUILD_JOBS} $(xcrun_find_tool dsymutil))
+
+        # Strip executables, shared libraries and static libraries in
+        # `host_install_destdir`.
+        find "${host_install_destdir}" \
+          -perm -0111 -or -name "*.a" -type f -print | \
```

Comment:
The previous version of this command included brackets. That doesn't seem to work (OSX 10.11.4); `find` returns too many files, including scripts and executables which are codesigned and can't be stripped.

Luckily we don't seem to need to brackets.